kernel/src/main/scala/cats/kernel/Order.scala

```diff
@@ -232,5 +232,7 @@ object Order extends OrderFunctions[Order] with OrderToOrderingConversion {
       override def toOrdering: Ordering[A] = ev
     }
 
-  def fromComparable[A <: Comparable[A]]: Order[A] = _ compareTo _
+  private type ContravariantComparable[A] = Comparable[? >: A]
+
+  def fromComparable[A <: ContravariantComparable[A]]: Order[A] = _ compareTo _
```

Comment:
seems to me like this should be implicit potentially, perhaps at a lower priority.

Reply:
@johnynek if an implicit parameter is added, would that break binary compatibility?

Reply:
I don't mean an implicit parameter, I mean `implicit def from comparable...`

This won't break JVM binary compatibility. But maybe we need to lower the priority of this implicit by putting it in a trait and extending here.